dsc/workflows/opencourseware.py

```diff
@@ -13,6 +14,234 @@
+            if (instructor_name := cls._format_instructor_name(instructor_details))
+        ]
+
+    @classmethod
+    def _format_instructor_name(cls, instructor_details: dict[str, str]) -> str:
+        """Format instructor name as 'Last, First Middle'.
+
+        Example:
+            Input: {"first_name": "Jane", "last_name": "Doe", "middle_initial": "A."}
+            Output: "Doe, Jane A."
+        """
+        if not (last_name := instructor_details.get("last_name")) or not (
+            first_name := instructor_details.get("first_name")
+        ):
+            return ""
+        instructor_name = (
+            f"{last_name}, {first_name} {instructor_details.get("middle_initial", "")}"
+        )
+        return instructor_name.strip()
+
+    @classmethod
+    def creativework_learningresourcetype(cls, source_metadata: dict) -> list[str]:
+        """Return the list of learning resource types.
+
+        Example:
+            Input: {"learning_resource_types": ["Lecture Notes", "Exams"]}
+            Output: ["Lecture Notes", "Exams"]
+        """
+        return source_metadata["learning_resource_types"]
+
+    @classmethod
+    def dc_subject(cls, source_metadata: dict) -> list[str]:
+        """Concatenate topic arrays into dash-separated strings.
+
+        Example:
+            Input: {"topics": [["Math", "Algebra"], ["Science", "Physics"]]}
+            Output: ["Math - Algebra", "Science - Physics"]
+        """
+        topics_list = [
+            " - ".join(topic_terms) for topic_terms in source_metadata["topics"]
+        ]
+        return list(filter(None, topics_list))
+
+    @classmethod
+    def dc_identifier_other(cls, source_metadata: dict) -> list[str]:
+        """Return a list of course identifiers, including formatted term/year.
+
+        Example:
+            Input: {
+                "primary_course_number": "6.001",
+                "extra_course_numbers": "18.01",
+                "term": "Fall",
+                "year": "2023"
+            }
+            Output: ["6.001", "6.001-Fall2023", "18.01"]
+        """
+        identifier_other_list = []
+        if primary_course_number := source_metadata["primary_course_number"]:
+            identifier_other_list.append(primary_course_number)
+            # format primary_course_number with term and year
+            identifier_other_list.append(
+                f"{primary_course_number}-{source_metadata["term"]}{source_metadata["year"]}"
+            )
+        if extra_course_numbers := source_metadata["extra_course_numbers"]:
+            identifier_other_list.append(extra_course_numbers)
+        return identifier_other_list
+
+    @classmethod
+    def dc_coverage_temporal(cls, source_metadata: dict) -> str:
+        """Return a string combining term and year.
+
+        Example:
+            Input: {"term": "Spring", "year": "2024"}
+            Output: "Spring 2024"
+        """
+        return " ".join([source_metadata["term"], source_metadata["year"]])
```

Comment:
Since `source_metadata["term"]` and `source_metadata["year"]` get joined a few times, you could consider whether it's worth doing it as an attribute

Reply:
Keeping as-is for now!